magento/pwa-studio
# Portal Component

## Description

This PR creates a new `Portal` component for rendering content into a DOM node that exists
outside of the DOM hierarchy of the parent / calling component.

The `Portal` component is a simple wrapper around [React Portals](https://reactjs.org/docs/portals.html) that provides a default target if needed.

We had this functionality already actually but the component was called `Modal` and it lead to confusion, especially after we introduced the `Dialog` component which has a "modal" mode.

In order to preserve backwards compatibility, we kept the `Modal` component around for now. It  exists as an alias for `Portal`. In Venia though, all instances of `Modal` have been replaced by `Portal`.

Another option we discussed was to have the `Modal` component compose the `Dialog` component and set its `isModal` prop to `true`. This may be a future improvement and is out of scope here.

## Related Issue





Closes PWA-590.

## Acceptance 



### Verification Stakeholders


@jimbo 

### Specification



### Verification Steps



All of the places where the new `Portal` component is used should be checked for regressions.
Please check that the following places function correctly on desktop and on mobile:

1. Filter Modal on the PDP
1. Edit an Item on the Cart Page
1. Edit Shipping Information on Checkout Page
1. Edit Shipping Method on Checkout Page
1. Edit Payment Information on Checkout Page

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
